packages/govuk-frontend/src/govuk/components/accordion/accordion.mjs

```diff
@@ -139,7 +139,10 @@ export class Accordion extends GOVUKFrontendComponent {
 
     const $sections = this.$module.querySelectorAll(`.${this.sectionClass}`)
     if (!$sections.length) {
-      return this
+      throw new ElementError({
+        componentName: 'Accordion',
+        identifier: `Sections (\`.${this.sectionClass}\`)`
```

Comment:
Are we alright moving all these to HTML still?

```suggestion
        identifier: `Sections (\`<div class="govuk-accordion__section">\`)`
```

Reply:
I'd be enclined to use HTML too (as well as for the other elements that may be missing) 😊

Reply:
Switched things to HTML where possible.

Section headings change tag depending on the `headingLevel` param (and I don't have time before I clock off to check how easy that'd be to use in generating the message).

Reply:
Yeah happy to add it, but guessing that promotes it to "JavaScript config" land so will cause issues 😣

What if the Nunjucks is rendered with `headingLevel: 2` but JS initialised with `headingLevel: 3`??

We could use [`Omit`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) to exclude it from the JavaScript config so it's available post-merge with the `data-*` attributes but can't be passed in, but that's something I've been looking at in https://github.com/alphagov/govuk-frontend/issues/4230

Or go ahead and update the JSDoc like you said but use `validateConfig()` to ban it

Reply:
I'm not sure we should forward Nunjucks config all the way down to the JavaScript component just for formatting an error message. That's also asking people using raw HTML to have that attribute set correctly, same for libraries implementing our templates in other languages that would need to update. I find it quite and ask for the benefit it brings.

A couple of other options:

1. Use a generic `hX` or `hN` name for the tag... might lead people to have actual `hX` tag in their markup
    ```
    `Section heading (\`<hX class="${this.sectionHeadingClass}">\`)`
    ```

2. Default to `h2`, at least they'll have a valid tag, but it may mess their heading hierarchy if they copy-paste
   ```
   `Section heading (\`<h2 class="${this.sectionHeadingClass}">\`)`
   ```
3. List the possible tag names, people will need to edit anyway
   ```
   `Section heading (\`<{h2,h3,h4,h5,h6} class="${this.sectionHeadingClass}">\`)`
   ```
4. Keep CSS for that one, 
    ```
    `Section heading (\`.${this.sectionHeadingClass}\`)`
    ```
    We could even highlight the name of the tag using the `:is()` selector
    ```
    `Section heading (\`:is(h2,h3,h4,h5,h6).${this.sectionHeadingClass}\`)`
    ```

1,2 and 3 seem quite prone to copy-pasting errors, so I'd lean towards 4 (in its simplest form to avoid one extra update), with 1 seeming the least worst of the HTML options if we prefer HTML.

Reply:
Yeah, I'd also favour just keeping it as is rather than wrangle with trying to get the exact right heading level. Will keep it as

````Section heading (\`.${this.sectionHeadingClass}\`)````